bin/BuildPackages.sh

```diff
@@ -86,6 +105,28 @@ then
   GAPDIR="$(pwd)"
   popd
 fi
+
+# check if coloring should be turned off
+if [[ "$#" -ge 1 ]]
+then
+  case "$1" in
+    --no-color|--no-colors|--no-colour|--no-colours|\
+    --nocolor|--nocolors|--nocolour|--nocolours)
+      COLOR=0
+      shift
+    ;;
+  esac
+fi
```

Comment:
I think it's a bad idea to do arg parsing this late, inside a function. Same for the check for gapicon.bmp -- all these checks should be the first thing this script does.

And I know that this means they won't be logged. That's fine by me -- anybody who wants to log them can log the output of this script to a file. As such, I am somewhat sceptical about the whole logging mechanism in here. I think a better way for us to handle this whole affair would be to split this file into two: one which simply knows how to build a single package, and one which calls the subcript as appropriate. Then, remove any logging code from the former script.